vms/platformvm/vm.go

```diff
@@ -152,7 +153,9 @@ func (vm *VM) Initialize(
 		return err
 	}
 
-	validatorManager := pvalidators.NewManager(chainCtx.Log, vm.Config, vm.state, vm.metrics, &vm.clock)
+	// Invariant: The context lock is never grabbed when holding the state lock.
+	var stateLock sync.RWMutex
```

Comment:
nit: I wonder if we could find a more expressive name for `stakeLock`. IIUC it protects the validatorSet as a whole. Should we rename it to `validatorSetLock`?